src/macaron/dependency_analyzer/java_repo_finder.py

```diff
@@ -150,9 +158,53 @@ def find_scm(pom: Element, tags: list[str]) -> tuple[Iterator[str], int]:
                 # Add the contents of the final tag
                 results.append(element.text.strip())
 
+    # Resolve any Maven properties within the results
+    results = _resolve_properties(pom, results)
+
     return iter(results), len(results)
 
 
+def _resolve_properties(pom: Element, values: list[str]) -> list[str]:
+    """Resolve any Maven properties found within the passed list of values.
+
+    Maven POM files have five different use cases for properties (see https://maven.apache.org/pom.html).
+    Only the two that relate to contents found elsewhere within the same POM file are considered here.
+    That is: ${project.x} where x can be a child tag at any depth, or ${x} where x is found at project.properties.x
+    Entries with rejected properties are removed from the value list.
+    """
+    resolved_values = []
+    print(f"\nValues: {values}\n")
+    for value in values:
+        replacements: list = []
+        # Calculate replacements
+        found = False
+        print(f"\nValue: {value}\n")
+        for match in re.finditer("\\$\\{[^}]+}", value):
+            found = True
+            text = match.group().replace("$", "").replace("{", "").replace("}", "")
+            print(f"Text: {text}")
+            if text.startswith("project."):
+                text = text.replace("project.", "")
+            else:
+                text = f"properties.{text}"
+            print(f"Final Text: {text}")
+            value_iterator, count = find_scm(pom, [text])
```

Comment:
For example, with the following pom.xml:
```
    ....
    <licenses>
        <license>
            <name>${project.licenses.license.name}</name>
            <url>https://example.example/license</url>
            <distribution>repo</distribution>
        </license>
    </licenses>
    <scm>
        <connection>
            ssh://git@hostname:port/owner/${project.licenses.license.name}.git
        </connection>
        <developerConnection>
            git@github.com:owner/project${javac.src.version}.git
        </developerConnection>
        <url>https://github.com/owner/project</url>
        <tag>example-0.0.1</tag>
    </scm>
    ....
```
When run, this code will cause a recursion error:
```
src/macaron/dependency_analyzer/java_repo_finder.py:162: in find_scm
    results = _resolve_properties(pom, results)
src/macaron/dependency_analyzer/java_repo_finder.py:191: in _resolve_properties
    value_iterator, count = find_scm(pom, [text])
src/macaron/dependency_analyzer/java_repo_finder.py:162: in find_scm
    results = _resolve_properties(pom, results)
src/macaron/dependency_analyzer/java_repo_finder.py:191: in _resolve_properties
    value_iterator, count = find_scm(pom, [text])
src/macaron/dependency_analyzer/java_repo_finder.py:162: in find_scm
    results = _resolve_properties(pom, results)
E   RecursionError: maximum recursion depth exceeded in __instancecheck__
!!! Recursion detected (same locals & position)
```

Reply:
I guess one way of addressing this is to make sure that we only try to resolve the reference once. I think the call to `_resolve_properties` could happen after the call to `find_scm` instead of calling `_resolve_properties` inside `find_scm`.